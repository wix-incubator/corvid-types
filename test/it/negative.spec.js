const compiler = require("../utils/compiler");
const listSubDirectories = require("../utils/listSubDirectories");
const writeTypingsHelper = require("../utils/writeTypingsHelper");
const testParser = require("../utils/testParser");

const getDynamicTypings = require("../utils/dynamicTypings");
const { declarations } = require("../../src").default;

const NEGATIVE_ROOT_PATH = "test/it/code-samples/negative";
const ignoredTests = [
  "test/it/code-samples/negative/pages-$w-dynamic-not-on-page"
];
const negativeRoots = listSubDirectories(NEGATIVE_ROOT_PATH).filter(
  subDirectory => !ignoredTests.includes(subDirectory)
);

describe("typescript - negative scenarios - configPaths flow", () => {
  it.each(negativeRoots)(
    "should fail compiling %s folder",
    async tsRootPath => {
      const { context, elementsMap, widgets, dependencies } = testParser.read(
        tsRootPath
      );

      const testTmpDirPath = writeTypingsHelper.dynamic(
        tsRootPath,
        context,
        getDynamicTypings({ elementsMap, widgets, dependencies })
      );

      expect(() => {
        compiler(`${testTmpDirPath}/tsconfig.json`);
      }).toThrowErrorMatchingSnapshot();
    }
  );
});

describe("typescript - negative scenarios - declarations flow", () => {
  it.each(negativeRoots)(
    "should fail compiling %s folder",
    async tsRootPath => {
      const { context, elementsMap, widgets, dependencies } = testParser.read(
        tsRootPath
      );

      const allTypes = await declarations[context]({
        elementsMap,
        widgets,
        dependencies
      });

      const testTmpDirPath = writeTypingsHelper.full(tsRootPath, allTypes);

      expect(() => {
        compiler(`${testTmpDirPath}/tsconfig.json`);
      }).toThrowErrorMatchingSnapshot();
    }
  );
});
