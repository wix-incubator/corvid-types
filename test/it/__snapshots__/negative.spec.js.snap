// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/backend-$w-module folder 1`] = `"$w-module.js (1,1): Cannot use namespace '$w' as a value."`;

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/backend-unknown-module folder 1`] = `"unknown.js (1,11): Cannot find name 'wixUnknownModule'."`;

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/pages-node-module folder 1`] = `"node.js (1,20): Cannot find module 'fs'.,node.js (2,22): Cannot find module 'path'."`;

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/pages-unknown-module folder 1`] = `"unknown.js (1,11): Cannot find name 'wixUnknownModule'."`;

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/public-$w folder 1`] = `"index.js (1,1): Cannot use namespace '$w' as a value."`;

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/public-node-module folder 1`] = `"node.js (1,20): Cannot find module 'fs'.,node.js (2,22): Cannot find module 'path'."`;

exports[`typescript - negative scenarios - configPaths flow should fail compiling test/it/code-samples/negative/public-unknown-module folder 1`] = `"unknown.js (1,11): Cannot find name 'wixUnknownModule'."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/backend-$w-module folder 1`] = `"$w-module.js (1,1): Cannot use namespace '$w' as a value."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/backend-unknown-module folder 1`] = `"unknown.js (1,11): Cannot find name 'wixUnknownModule'."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/pages-node-module folder 1`] = `"node.js (1,20): Cannot find module 'fs'.,node.js (2,22): Cannot find module 'path'."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/pages-unknown-module folder 1`] = `"unknown.js (1,11): Cannot find name 'wixUnknownModule'."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/public-$w folder 1`] = `"index.js (1,1): Cannot use namespace '$w' as a value."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/public-node-module folder 1`] = `"node.js (1,20): Cannot find module 'fs'.,node.js (2,22): Cannot find module 'path'."`;

exports[`typescript - negative scenarios - declarations flow should fail compiling test/it/code-samples/negative/public-unknown-module folder 1`] = `"unknown.js (1,11): Cannot find name 'wixUnknownModule'."`;
